tcms/testruns/static/testruns/js/get.js

```diff
@@ -1,5 +1,88 @@
+let executionStatuses;
+
 $(document).ready(() => {
 
     $('.bootstrap-switch').bootstrapSwitch();
 
+    const testRunId = $('#test_run_pk').data('pk')
+
+    jsonRPC('TestExecutionStatus.filter', {}, data => {
+        executionStatuses = data
+        drawPercentBar(testRunId)
+    })
 })
+
+function drawPercentBar(testRunId) {
+
+    jsonRPC('TestExecution.filter', { 'run_id': testRunId }, testExecutions => {
+
+        let positiveCount = 0;
+        let negativeCount = 0;
+        let allCount = testExecutions.length;
+        let statusCount = {}
+        executionStatuses.forEach(s => statusCount[s.name] = { count: 0, id: s.id })
+
+        testExecutions.forEach(testExecution => {
+            const executionStatus = executionStatuses.find(s => s.id === testExecution.status_id)
+
+            if (executionStatus.weight > 0) {
+                positiveCount++
+            } else if (executionStatus.weight < 0) {
+                negativeCount++
+            }
+
+            statusCount[executionStatus.name].count++
+        })
+
+        renderProgressBars(positiveCount, negativeCount, allCount)
+        renderCountPerStatusList(statusCount)
+    })
+}
+
+function renderProgressBars(positiveCount, negativeCount, allCount) {
+
+    const positivePercent = +(positiveCount / allCount * 100).toFixed(2)
```

Comment:
Why the `+` sign ?

Reply:
to convert the var to number, so that we can call `toFixed`